### PR TITLE
Re-jig of menu positioning

### PR DESCRIPTION
**Pull request by julianjelfs**
This is mostly based on this lib: https://github.com/simonwep/nanopop

But I had to tweak it to allow it to force a selection in the case of a complete miss while truncating the height / width. 

So in the case of the crypto selector for example it will now try to position the menu, find that there is no variant that fits and then recurse with a force flag to just take the preferred alignment and _make_ it fit.

### CODE REVIEW

File: frontend/app/src/components/MenuIcon.svelte [hunk 1]
```diff
@@ -57,7 +56,8 @@
     }
 
     .menu {
-        position: absolute;
+        // position: absolute;
```
Comment (megrogan) [inline frontend/app/src/components/MenuIcon.svelte:59]: Does this need to hang around?

File: frontend/app/src/components/home/BigDebugMenu.svelte [hunk 2]
```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
```
Comment (megrogan) [inline frontend/app/src/components/home/BigDebugMenu.svelte:1]: Does this need to hang around?

File: frontend/app/src/stores/menu.ts [hunk 3]
```diff
@@ -31,41 +26,40 @@ function close(menu: HTMLElement | undefined): HTMLElement | undefined {
...
-                menu.style.setProperty("--override-height", `${dim.h}px`);
+            if (centered && get(mobileWidth)) {
+                positionInCenter(menu);
+            } else {
+                const pos = reposition(triggerEl, menu, {
+                    position: `${position}-${align}`,
+                    margin: gutter,
+                });
+                if (!pos) {
+                    reposition(triggerEl, menu, {
```
Comment (megrogan) [inline frontend/app/src/stores/menu.ts:56]: Couldn't this retry happen within reposition itself? Then also shared by tooltip.

**Review by megrogan — DISMISSED** [review]

File: frontend/app/src/components/home/BigDebugMenu.svelte (hunk 2 above)
Comment (julianjelfs) [reply frontend/app/src/components/home/BigDebugMenu.svelte:1]: I think it's quite useful for testing so I wouldn't mind leaving it. Shouldn't do any harm.

File: frontend/app/src/components/MenuIcon.svelte (hunk 1 above)
Comment (julianjelfs) [reply frontend/app/src/components/MenuIcon.svelte]: nope. gone.

File: frontend/app/src/stores/menu.ts (hunk 3 above)
Comment (julianjelfs) [reply frontend/app/src/stores/menu.ts:56]: yes I guess so. I will change it.

**Review by megrogan — APPROVED** [review]

**Pull request merged**